src/mscorlib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -695,7 +695,7 @@ private static Boolean Lex(DS dps, ref __DTString str, ref DateTimeToken dtok, r
                                     }
                                     else
                                     {
-                                        result.SetFailure(ParseFailureKind.Format, nameof(SR.Format_BadDateTime), null);
+                                        result.SetFailure(ParseFailureKind.FormatWithParameter, nameof(SR.Format_BadDateTime), new string(str.Value));
```

Comment:
> `new string(str.Value)`

This allocation will slow down the `TryParse` methods. Can the allocation be delayed until the exception message actually needs to be created?

Reply:
You're right, I was actually worried about these allocations. I'll look into your suggestion.